structuralcodes/materials/concrete/_concreteEC2_2023.py

```diff
@@ -0,0 +1,195 @@
+"""The concrete class for EC2 2023 Concrete Material."""
+
+import typing as t
+
+from structuralcodes.codes import ec2_2023
+from ._concrete import Concrete
+
+
+# TODO: complete the documentation and discuss if properties/functions
+# are considered proper for the framework
+class ConcreteEC2_2023(Concrete):  # noqa: N801
+    """Concrete implementation for EC2 2023 Concrete."""
+
+    # Inherent concrete properties
+    _kE: t.Optional[float] = None  # noqa: N815
+    _concrete_dev_class: t.Optional[str] = None
+    _gamma_C: t.Optional[float] = None  # noqa: N815
+
+    # Computed attributes
+    _fcm: t.Optional[float] = None
+    _fctm: t.Optional[float] = None
+    _Ecm: t.Optional[float] = None
+
+    # TODO: do we agree with the optional keyword arguments here?
+    def __init__(
+        self,
+        fck: float,
+        name: t.Optional[str] = None,
+        density: float = 2400.0,
+        kE: float = 9500,
+        concrete_dev_class: str = 'CS',
+        gamma_C: float = 1.5,
+        existing: bool = False,
+    ):
+        """Initializes a new instance of Concrete for MC 2010.
+
+        Args:
+            fck (float): Characteristic strength in MPa if concrete is not
+                existing.
+
+        Keyword Args:
+            name (str): A descriptive name for concrete
+            density (float): Density of material in kg/m3 (default: 2400)
+            kE (float): coefficient relating aggregates.
+            concrete_dev_class (str, optional): default is CS
+            gamma_C (float, optioanl): partial factor of concrete
+                (default is 1.5)
+            existing (bool, optional): The material is of an existing structure
+                (default: False)
+        """
+        if name is None:
+            name = f'C{round(fck):d}'
+        super().__init__(
+            fck=fck, name=name, density=density, existing=existing
+        )
+        self._kE = kE
+        self._concrete_dev_class = concrete_dev_class
+        self._gamma_C = gamma_C
+
+        # TODO: implement this method
+        def _reset_attributes(self):
+            # We only need to reset computed properties
+            self._fcm = None
+            self._fctm = None
+            self._Ecm = None
+
+        @property
+        def fcm(self) -> float:
+            """Returns the mean strength of concrete.
+
+            Returns:
+                float: The mean compressive strength in MPa.
+            """
+            if self._fcm is not None:
+                return self._fcm
+            self._fcm = ec2_2023.fcm(self.fck)
+            return self._fcm
```

Comment:
this lines can be simply condensed in one:
`return ec2_2023.fcm(self.fck)`

Reply:
Mmm! Just wondering, I guess that we can do something as follows:
```
@property
def fcm(self) -> float:
    """Returns the mean strength of concrete.

    Returns:
        float: The mean compressive strength in MPa.
    """
    self._fcm = self._fcm or ec2_2023.fcm(self.fck)
    return self._fcm
```

I guess that we may like to store the `self._fcm` value just in case that the parent class calls `_reset_attribules(self)`, don't we?

Reply:
I agree with your suggestion, @DanielGMorenaFhecor

Reply:
I am not sure I am getting it correctly...
In my original proposal I did the following reasoning:
- by default we have some attributes that have some values based on equations of the code (like fcm).
- the user can always override these defaults.

That's way I did the following:
self._fcm as default is None. The getter return the code result if it is None. If the user override this, the getter will return the user-set value. When we reset is because we want to go back to code-equation computations.
I wasn't thinking at those internal values for caching results.

As far as I can understand the code you are proposing cache the result, so we (and the user) no longer can keep track on what attributes have been override and which have not been override...

I don't know if I am making sense...

Reply:
The way this is written in the commited code proposal seems to follow the original reasoning described by @talledodiego above, however written slightly different than in e.g. `ConcreteMC2010` where the value from the code equation is not stored in the internal attribute. I suggest to keep the proposal as it is 😃